src/common/geometry.cc

```diff
@@ -0,0 +1,71 @@
+// BoB robotics includes
+#include "common/geometry.h"
+#include "common/macros.h"
+
+// Third-party includes
+#include "third_party/units.h"
+
+// Eigen
+#include <Eigen/Geometry>
+
+// Standard C includes
+#include <cmath>
+
+// Standard C++ includes
+#include <vector>
+
+namespace BoBRobotics {
+double
+distance2D(const Eigen::Vector2d &p1, const Eigen::Vector2d &p2)
+{
+    return hypot(p2.x() - p1.x(), p2.y() - p1.y());
+}
+
+
+double min(const Eigen::Matrix2d &line, int index)
+{
+    return line.col(index).minCoeff();
+}
+
+double max(const Eigen::Matrix2d &line, int index)
+{
+    return line.col(index).maxCoeff();
+}
+
+bool
+calculateIntersection(Eigen::Vector2d &point, const Eigen::Matrix2d &line1Points, const Line2 &line2)
+{
+    const auto lineEq1 = Line2::Through(line1Points.row(0), line1Points.row(1));
+    point = lineEq1.intersection(line2);
+
+    // Margin of error for floating-point numbers
+    constexpr double tol = std::numeric_limits<double>::epsilon();
+
+    // Check that the point is on line 1
+    const bool a = min(line1Points, 0) <= point(0) + tol;
```

Comment:
I think the classic test for this is to project i.e. dot product ``point - line1Points.row(0)`` with the direction of line1 (which you should be able to get from the normal of ``lineEq1``) and it's on the line if it's >= 0 and less than the length of line1